### PR TITLE
Feature engage

### PR DESCRIPTION
**Pull request by Odkhuu217**
feature-engage

### CODE REVIEW

**Comment by ganzorig** [comment]
@Odkhuu217 edit hiih geheer buh yum ni hooson bgad bna. Ashiglaagui code bas haragdaad bna. Refactor neleen hiimeer bna. Commit message uudee uguulbereer oilgomjtoi bichmeer bna. Zarim neg commentuudiig file dotor ni bichie.

File: src/modules/engage/components/FormBase.js [hunk 1]
```diff
@@ -1,194 +1,45 @@
...
     super(props);
 
-    const message = props.message || {};
-
-    this.state = { fromUser: message.fromUserId || '' };
-
-    // binds
-    this.generateDoc = this.generateDoc.bind(this);
+    this.state = {
+      a: 123
```
Comment (ganzorig) [inline src/modules/engage/components/FormBase.js]: ene haana ashiglagdaj bga state we? ashiglahgui bol ustgaad ashiglaj bga bol oilgomjtoi nereer bichih

File: src/modules/engage/components/FormBase.js [hunk 2]
```diff
@@ -1,194 +1,45 @@
...
-  }
-
-  renderContent() {
-    return null;
-  }
-
   renderTitle() {
     const { kind } = this.props;
-
+    let title = 'Visitor auto message';
```
Comment (ganzorig) [inline src/modules/engage/components/FormBase.js]: let iig ashiglahgui bwal zugeer bna. Eniig jisheelbel neg func bolgood 2 nuhtsuluu shalgaad return string butsaagaad defaultaaraa Visitor auto message eniig butsaadag bolgowol bolchih sanagdaad bna

File: src/modules/engage/components/FormBase.js [hunk 3]
```diff
@@ -1,194 +1,45 @@
...
-    return (
-      <Wrapper
-        header={<Wrapper.Header breadcrumb={breadcrumb} />}
-        footer={actionFooter}
-        content={this.renderContent()}
-        leftSidebar={sidebar}
-      />
-    );
+    const breadcrumb = [{ title: 'Engage', link: '/engage' }, { title: title }];
+    return breadcrumb;
```
Comment (ganzorig) [inline src/modules/engage/components/FormBase.js]: shuud arraygaa butsaachij bolohoor bna zaawal huwisagchid uguhgui

File: src/modules/engage/components/MessengerForm.js [hunk 4]
```diff
@@ -65,81 +79,75 @@ class MessengerForm extends Component {
...
+            >
+              <option />
+              {this.props.brands.map(b => (
+                <option key={b._id} value={b._id}>
+                  {b.name}
+                </option>
+              ))}
+            </FormControl>
+          </FormGroup>
+          {kind}
```
Comment (ganzorig) [inline src/modules/engage/components/MessengerForm.js]: End jisheelbel kind ni yumuu eswel hooson string hewleed bga gesen ug shd

File: src/modules/engage/components/MessengerForm.js [hunk 5]
```diff
@@ -2,58 +2,72 @@ import React, { Component } from 'react';
...
+  changeUser(fromUser) {
+    this.setState({ fromUser });
+    this.props.changeMessenger('fromUser', fromUser);
   }
 
-  renderMessageType(messenger) {
-    if (this.props.showMessengerType) {
-      return (
+  render() {
+    let kind = '';
```
Comment (ganzorig) [inline src/modules/engage/components/MessengerForm.js]: ene hesgiig neg func bolgochihwol zuger bna iimerhuu yum neleen haragdaj bsan. Jisheelbel renderKind(){
 if(hasKind) { return (< FormGroup >content< FormGroup >)}
return null;
}
gechwel let ashiglahguigeer shiidej bolohoor haragdaj bna.

File: src/modules/engage/components/SegmentsForm.js [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+  create: PropTypes.func.isRequired,
+  segment: PropTypes.object,
+  headSegments: PropTypes.array.isRequired,
+  count: PropTypes.func.isRequired,
+  total: PropTypes.object
+};
+
+class SegmentsForm extends Component {
+  static generateRandomColorCode() {
+    return `#${Math.random()
```
Comment (ganzorig) [inline src/modules/engage/components/SegmentsForm.js]: ene random color yund ashiglaj bna

File: src/modules/engage/components/VisitorForm.js [hunk 7]
```diff
@@ -1,182 +1,142 @@
...
-  MESSAGE_KINDS
-} from 'modules/engage/constants';
-
 import { Wrapper } from 'modules/layout/components';
-
-import FormBase from './FormBase';
 import MessengerForm from './MessengerForm';
+import { StepWrapper, TitleContainer } from './step/style';
+import { FormControl } from 'modules/common/components';
+import Steps from './step/Steps';
```
Comment (ganzorig) [inline src/modules/engage/components/VisitorForm.js]: Ene importuudiig negtgeh

File: src/modules/engage/components/step/SegmentStep.js [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+`;
+
+const propTypes = {
+  changeSegment: PropTypes.func,
+  segments: PropTypes.array,
+  counts: PropTypes.object,
+  segmentPush: PropTypes.func
+};
+
+class Step2 extends Component {
```
Comment (ganzorig) [inline src/modules/engage/components/step/SegmentStep.js]: step1, 2,3 uudiig file name tei ni ijilhen bolgoson ni oilgomjtoi bha

File: src/modules/engage/components/step/Step.js [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+    } else if (save) {
+      return (
+        <Button.Group>
+          <Button
+            btnStyle="warning"
+            size="small"
+            icon="ios-arrow-forward"
+            onClick={e => save('draft', e)}
+          >
+            Save Draft
```
Comment (ganzorig) [inline src/modules/engage/components/step/Step.js]: step 3 ruu orohoor ene Save & Live SAve & Draft gesen buttonguud haragdahgui bhin dandaa save l gej garaad bhin

File: src/modules/engage/components/step/style.js [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+
+const StepStatus = styled.div`
+  margin-top: 20px;
+`;
+
+const FlexItem = styled.div`
+  display: flex;
+  height: 100%;
+  flex: ${props => (props.count ? props.count : 1)};
+  ${props => {
```
Comment (ganzorig) [inline src/modules/engage/components/step/style.js]: neg props dotor hiichij bolohoor haragdaj bna

File: src/modules/engage/components/step/style.js [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+
+const Show = styled.div`
+  display: ${props => (props.show ? 'block' : 'none')};
+`;
+
+const Divider = styled.div`
+  width: 1px;
+  background: ${colors.borderPrimary};
+  height: 100%;
+  margin: 0 10px;
```
Comment (ganzorig) [inline src/modules/engage/components/step/style.js]: ene margin iin 0 bolgochih

File: src/modules/engage/components/SegmentsForm.js (hunk 6 above)
Comment (Odkhuu217) [reply src/modules/engage/components/SegmentsForm.js]: stepiin create form heseg deer segment nemehed default colort

**Comment by ganzorig** [comment]
@Odkhuu217 console deer 2 warning garj bna. Bas console loguudaa arilgaarai. Messenger songood hadgalaad butsaad edit hiih orohoor hooson bgad bna

**Pull request closed without merging**